pulp_smash/tests/platform/cli/test_pulp_manage_db.py

```diff
@@ -0,0 +1,124 @@
+# coding=utf-8
+"""Tests for the ``pulp-manage-db`` executable.
+
+``pulp-manage-db`` should only run when :data:`REQUIRED_SERVICES` are running
+and when :data:`CONFLICTING_SERVICES` are stopped. See:
+
+* `Pulp #2186 <https://pulp.plan.io/issues/2186>`_
+* `Pulp Smash #487 <https://github.com/PulpQE/pulp-smash/issues/487>`_
+"""
+import inspect
+import unittest
+
+from pulp_smash import cli, config, exceptions, selectors, utils
+
+REQUIRED_SERVICES = frozenset(('mongod',))
+"""If any of these services are stopped, ``pulp-manage-db`` will abort."""
+
+CONFLICTING_SERVICES = frozenset((
+    'pulp_celerybeat',
+    'pulp_resource_manager',
+    'pulp_workers',
+))
+"""If any of these services are running, ``pulp-manage-db`` will abort."""
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Log in."""
+    utils.pulp_admin_login(config.get_config())
+
+
+def tearDownModule():  # pylint:disable=invalid-name
+    """Reset Pulp, in case one of the test cases breaks Pulp."""
+    utils.reset_pulp(config.get_config())
+
+
+class BaseTestCase(unittest.TestCase):
+    """An abstract base class for the test cases in this module."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Maybe skip this test case."""
+        if inspect.getmro(cls)[0] == BaseTestCase:
```

Comment:
We are adding base test cases classes as we increase the coverage. That makes me thing if we should create a base class for the base test case classes where the logic to skip will not be repeated over and over again.

Off course this is not required on this PR but is something that I wanted to start a conversation about.